pkg/local_object_storage/shard/put.go

```diff
@@ -38,9 +52,18 @@ func (s *Shard) Put(prm PutPrm) (PutRes, error) {
 		return PutRes{}, ErrReadOnlyMode
 	}
 
-	data, err := prm.obj.Marshal()
-	if err != nil {
-		return PutRes{}, fmt.Errorf("cannot marshal object: %w", err)
+	var data []byte
+	var err error
+	if prm.binSet {
+		data = prm.objBin
+	} else {
+		data, err = prm.obj.Marshal()
+		if err != nil {
+			return PutRes{}, fmt.Errorf("cannot marshal object: %w", err)
+		}
+		// TODO: currently, we don't need to calculate prm.hdrLen in this case.
```

Comment:
todo with no issue?

Reply:
how can you be sure? we had a work once where the dev searched for such lost cases

Reply:
> how can you be sure?

this is what git is made for

Reply:
> git is made for

searching for `TODO` strings? my point is that `TODO` is about smth that _should_ be done. good to have a corresponding issue then. otherwise that is not `TODO`

Reply:
To me it's not even a TODO, just a remark on the specifics of the current code. It's nothing actionable at this point.